### PR TITLE
Factor out Usernames component for formatting lists of users

### PR DESCRIPTION
**Pull request by chromakode**
This component handles italicizing your own username and coloring broken usernames in a consistent manner (for use in displaying TLFs, titles, etc).

:eyeglasses: @keybase/react-hackers

### CODE REVIEW

File: shared/common-adapters/usernames.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import type {Props} from './usernames'
+
+export default class Usernames extends Component<void, Props, void> {
+  render () {
+    const {type, users, style} = this.props
+
+    return (
+      <Box style={{...globalStyles.flexBoxRow, flexWrap: 'wrap'}}>
+        {users.map((u, i) => {
+          const userStyle = Object.assign({}, style)
```
Comment (chrisnojima) [inline shared/common-adapters/usernames.js]: Prefer {...} over assign

**Comment by keybase-ci-visdiff** [comment]
:mag_right: :zap: These commits introduced some visual changes:
 * Changed: **Files-Popup_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Popup_Public-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Popup_Public-b52415ca3696.png)  
   Files-Popup_Public comparison
 * Changed: **Files-Popup_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Popup_Private-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Popup_Private-b52415ca3696.png)  
   Files-Popup_Private comparison
 * Changed: **Files-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Normal_Public-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Normal_Public-b52415ca3696.png)  
   Files-Normal_Public comparison
 * Changed: **Files-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Normal_Private-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464304980453-cc55233461f5-b52415ca3696/Files-Normal_Private-b52415ca3696.png)  
   Files-Normal_Private comparison

**Comment by chrisnojima** [comment]
:+1: % c

**Comment by keybase-ci-visdiff** [comment]
:mag_right: :zap: These commits introduced some visual changes:
 * Changed: **Files-Popup_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Popup_Public-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Popup_Public-7f382b5d1bea.png)  
   Files-Popup_Public comparison
 * Changed: **Files-Popup_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Popup_Private-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Popup_Private-7f382b5d1bea.png)  
   Files-Popup_Private comparison
 * Changed: **Files-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Normal_Public-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Normal_Public-7f382b5d1bea.png)  
   Files-Normal_Public comparison
 * Changed: **Files-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Normal_Private-cc55233461f5.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464307259901-cc55233461f5-7f382b5d1bea/Files-Normal_Private-7f382b5d1bea.png)  
   Files-Normal_Private comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: :zap: These commits introduced some visual changes:
 * Changed: **Menubar-Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Menubar-Truncated-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Menubar-Truncated-365387f1e05d.png)  
   Menubar-Truncated comparison
 * Changed: **Menubar-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Menubar-Normal-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Menubar-Normal-365387f1e05d.png)  
   Menubar-Normal comparison
 * Changed: **Folders_TLF-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Folders_TLF-Normal-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Folders_TLF-Normal-365387f1e05d.png)  
   Folders_TLF-Normal comparison
 * Changed: **Files-Popup_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Popup_Public-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Popup_Public-365387f1e05d.png)  
   Files-Popup_Public comparison
 * Changed: **Files-Popup_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Popup_Private-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Popup_Private-365387f1e05d.png)  
   Files-Popup_Private comparison
 * Changed: **Files-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Normal_Public-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Normal_Public-365387f1e05d.png)  
   Files-Normal_Public comparison
 * Changed: **Files-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Normal_Private-7cfb44a382a4.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464374827763-7cfb44a382a4-365387f1e05d/Files-Normal_Private-365387f1e05d.png)  
   Files-Normal_Private comparison

**Pull request merged**